app/scripts/actions/helpers.js

```diff
@@ -1,151 +1,67 @@
   if (body.name) error = body.name;
   if (body.message) error += `${(error ? ': ' : '')}${body.message}`;
   return error;
-}
-
-export const get = function (config, callback) {
-  request.get(config, (error, resp, body) => {
-    if (error) {
-      return callback(error);
-    } else if (+resp.statusCode >= 400) {
-      return callback(new Error(formatError(resp, body)));
-    }
-    return callback(null, body);
-  });
 };
 
-export const post = function (config, callback) {
-  request.post(config, (error, resp, body) => {
-    error = error || body.errorMessage;
-    if (error) {
-      return callback(error);
-    } else if (+resp.statusCode >= 400) {
-      return callback(new Error(formatError(resp, body)));
-    } else {
-      return callback(null, body);
-    }
-  });
-};
+export const getError = (response) => {
+  const { request, body } = response;
+  let error;
 
-export const put = function (config, callback) {
-  request.put(config, (error, resp, body) => {
-    error = error || body && body.errorMessage || body && body.detail || null;
-    if (error) {
-      return callback(error);
-    } else if (+resp.statusCode >= 400) {
-      return callback(new Error(formatError(resp, body)));
-    } else {
-      return callback(null, body);
-    }
-  });
-};
+  // TODO: is this still relevant?
+  if (request.method === 'DELETE' || request.method === 'POST') {
+    error = body.errorMessage;
+  } else if (request.method === 'PUT') {
+    error = body && body.errorMessage || body && body.detail;
+  }
 
-export const del = function (config, callback) {
-  request.del(config, (error, resp, body) => {
-    error = error || body.errorMessage;
-    if (error) {
-      return callback(error);
-    } else if (+resp.statusCode >= 400) {
-      return callback(new Error(formatError(resp, body)));
-    } else {
-      return callback(null, body);
-    }
-  });
+  if (error) return error;
+
+  error = new Error(formatError(response, body));
+  return error;
 };
 
-export const addRequestAuthorization = (config, getState) => {
-  let token = getProperty(getState(), 'api.tokens.token');
+export const addRequestAuthorization = (config, state) => {
+  let token = getProperty(state, 'api.tokens.token');
   if (token) {
-    config.headers = config.headers || {};
-    config.headers.Authorization = 'Bearer ' + token;
+    config.headers = Object.assign({}, config.headers, {
+      Authorization: `Bearer ${token}`
+    });
   }
-  return config;
 };
 
-export const configureRequest = function (params, body) {
-  let config;
-  if (typeof params === 'string') {
-    config = {
-      url: url.resolve(root, params)
-    };
-  } else if (params.url) {
-    config = params;
-  } else {
-    throw new Error('Must include a url with request');
-  }
-  config.json = true;
+export const configureRequest = (params = {}) => {
+  let config = params;
 
-  if (body && typeof body === 'object') {
-    config.body = body;
+  if (!config.url && !config.path) {
```

Comment:
This may be a bit easier to read, but completely up to you. This just gets rid of the need for a second `if (config.path)`
```nodejs
   if (!config.url || !config.path) {
   ...
    if (typeof config.path !== 'string') {
```

Reply:
`if (!config.url || !config.path) {` would throw an error if you didn't have one or the other though, but it is valid to include either `url` or `path`.